src/reducers/tabs.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Tabs reducer
+ * @module reducers/tabs
+ */
+
+import * as I from "immutable";
+import makeRecord from "../utils/makeRecord";
+import { prefs } from "../utils/prefs";
+
+import type { List } from "immutable";
+import type { Action } from "../actions/types";
+import type { Record } from "../utils/makeRecord";
+
+type Tab = {
+  id: string,
+  title: string,
+  tooltip?: string
+};
+
+type TabList = List<Tab>;
```

Comment:
Since we do not use movetab unless it is a hidden tab (and in that case, always moving it to index 0), i am wondering how important moving tabs is? Perhaps, if the uniqueness of the ID is more important, we can leverage orderedSet instead, with unique keys (the id being the key)? Since the id is not data being used in the tab itself, and to identify it in the group, this might make a few things easier.

Reply:
yeah your right we do not do much with move tabs, and i'm in favor of removing it. maybe add tabs can handle moving if a tab already exists. An orderedSet might be cool, will look into it a bit more.